torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -790,6 +790,7 @@ def reset_sharded_param(self):
         if updated_local_tensor:
             # Only change the local tensor object if needed
             self.sharded_param._local_tensor = local_tensor[: self.sharded_size[0]]
+        self._sharding_spec = self.sharded_param._spec
```

Comment:
Question. instead of cashing `self._sharding_spec` would it make sense to have it be a property that always just queries `self.sharded_param._spec`

Reply:
That sounds good!

Reply:
took a quick look -- this will require some refactoring, so I will defer this to a separate PR